### PR TITLE
Add get_output_file and get_error_file

### PR DESCRIPTION
**Pull request by wch**
This adds methods `get_output_file()` and `get_error_file()`.

I also implemented a function called `full_path()` that returns the full path to a file. Sort of like `normalizePath()`, but it doesn't follow symlinks, and it works even if the file does not exist.

### CODE REVIEW

**Comment by gaborcsardi** [comment]
Thanks! As for `full_path`, can't we just do `file.path(getwd(), path.expand(path))` for a relative path?

**Comment by wch** [comment]
`path.expand` only resolves `~` and `~username` and returns an absolute path in those cases, so I don't think that will work.

File: R/process.R [hunk 1]
```diff
@@ -186,6 +186,12 @@ NULL
 #' It returns a character vector. This will return content only if
 #' `stderr="|"` was used. Otherwise, it will throw an error.
 #'
+#' \code{$get_output_file()} returns the value of the `stdout` argument
+#' provided when the process object was created.
```
Comment (gaborcsardi) [inline R/process.R]: Well, that's not technically correct, because we expand and normalize the path, right?

File: R/process.R [hunk 2]
```diff
@@ -186,6 +186,12 @@ NULL
...
+#'
+#' \code{$get_error_file()} returns the value of the `stderr` argument
+#' provided when the process object was created.
```
Comment (gaborcsardi) [inline R/process.R]: Same here.

**Comment by gaborcsardi** [comment]
Thanks! I left some small comments.

File: R/process.R (hunk 1 above)
Comment (wch) [inline R/process.R]: Oops, that was the original behavior, but it has changed. I'll update it.

File: R/utils.R [hunk 3]
```diff
@@ -25,3 +25,76 @@ last_char <- function(x) {
...
 }
+
+# Given a filename, return an absolute path to that file. This has two important
+# differences from normalizePath(). (1) The file does not need to exist, and (2)
+# the path is merely absolute, whereas normalizePath() returns a canonical path,
+# which resolves symbolic links, gives canonical case, and, on Windows, may give
+# short names.
+#
+# On Windows, the returned path includes the drive ("C:") or network server
+# ("//myserver"). Only "/" is supported as a path separator (no backslashes).
```
Comment (wch) [inline R/utils.R]: Oh yeah, that's a pretty simple fix.

**Comment by wch** [comment]
OK, this should be good to go now.

**Comment by gaborcsardi** [comment]
Thanks!

**Pull request merged**